### PR TITLE
2665: Fix error in opening deep links in iOS

### PR DESCRIPTION
**Pull request by LeandraH**
### Short description

When trying to open a deep link in a different city, the inAppBrowser is opened twice in iOS, and the user ends up being sent to the dashboard page with an error in the Snackbar and in Sentry.

### Proposed changes

When redirecting to a deep link, we give it a little timeout first to allow the inAppBrowser to start. To allow for older devices, we also start an interval to keep trying to open the inAppBrowser. We don't check if the first method with the timeout worked before attempting a second time. I now added a little check to see if the inAppBrowser is already open before trying to open it again.

### Side effects

Hopefully nothing.

### Resolved issues

Fixes: #2665

### CODE REVIEW

**Review by f1sh1918 — APPROVED** [review]
Tested on ios.
Works nice 👍

File: native/src/components/RedirectContainer.tsx [hunk 1]
```diff
@@ -30,8 +31,11 @@ const RedirectContainer = ({ route, navigation }: RedirectContainerProps): React
 
   useEffect(() => {
     // To support potentially older devices taking longer we setup a separate interval to retry the navigation
-    const interval = setInterval(() => {
-      navigateToDeepLink(url)
+    const interval = setInterval(async () => {
+      const noInAppBrowserOpenYet = !(await InAppBrowser.isAvailable())
+      if (noInAppBrowserOpenYet) {
+        navigateToDeepLink(url)
+      }
```
Comment (steffenkleinle) [inline native/src/components/RedirectContainer.tsx:38]: :wrench: I find the current naming hard to understand, perhaps the following (a positive wording) would be better?
```suggestion
      const urlRedirected = await InAppBrowser.isAvailable()
      if (!urlRedirected) {
        navigateToDeepLink(url)
      }
```

Comment (steffenkleinle) [reply native/src/components/RedirectContainer.tsx:38]: :question: why does `InAppBrowser.isAvailable()` indicate whether it is already open? According to the docs:
> Detect if the device supports this plugin.

AFAICS, it only tells whether we are able to open a link in general. What am I missing?

**Review by steffenkleinle — APPROVED** [review]

File: native/src/components/RedirectContainer.tsx (hunk 1 above)
Comment (LeandraH) [reply native/src/components/RedirectContainer.tsx:38]: You're right, I misunderstood what it did! The code that I wrote is completely useless, it just doesn't run the interval on devices that can open the `inAppBrowser`, as in it never does. Good catch, thanks!

**Pull request merged**